src/modules/adaptors/Morpho/MorphoAaveV3ATokenP2PAdaptor.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.16;
+
+import { BaseAdaptor, ERC20, SafeTransferLib } from "src/modules/adaptors/BaseAdaptor.sol";
+import { IMorpho } from "src/interfaces/external/Morpho/IMorpho.sol";
+import { MorphoRewardHandler } from "src/modules/adaptors/Morpho/MorphoRewardHandler.sol";
+
+/**
+ * @title Morpho Aave V3 aToken Adaptor
+ * @notice Allows Cellars to interact with Morpho Aave V3 positions.
+ * @author crispymangoes
+ */
+contract MorphoAaveV3ATokenP2PAdaptor is BaseAdaptor, MorphoRewardHandler {
+    using SafeTransferLib for ERC20;
+
+    //==================== Adaptor Data Specification ====================
+    // adaptorData = abi.encode(address underlying)
+    // Where:
+    // `underlying` is the ERC20 position this adaptor is working with
+    //================= Configuration Data Specification =================
+    // configurationData = abi.encode(uint256 maxIterations);
+    // Where:
+    // `maxIterations` is some number greater than 0 but less than MAX_ITERATIONS,
+    // allows assets to be P2P matched.
+    //====================================================================
+
+    uint256 internal constant MAX_ITERATIONS = 10;
```

Comment:
Where did these iteration values come from?

Reply:
Is this in place for when maxGas caps won't necessarily guarantee adequate APR because the Cellar may not match with a P2P APY?

Reply:
So in their contracts, they have a comment saying a good iterations value is 4, so I somewhat arbitrarily chose 10 as a max. I don't think we would be getting much value from using a value greater than 4, but the main reason why we have a cap is so that strategists can't gas grief our relayer by using a stupid big iterations value.